client/core/core.go

```diff
@@ -2614,12 +2607,27 @@ func (c *Core) registerHD(dc *dexConnection, crypter encrypt.Crypter) (*incomple
 	}
 
 	creds := c.creds()
-	encKey, acctPubB, err := dc.acct.setupCryptoV2(creds, crypter)
-	if err != nil {
-		return nil, false, newError(acctKeyErr, "setupCryptoV2 error: %v", err)
-	}
+	for {
+		encKey, acctPubB, err := dc.acct.setupCryptoV2(creds, crypter)
+		if err != nil {
+			return nil, false, newError(acctKeyErr, "setupCryptoV2 error: %v", err)
+		}
+
+		incompleteReg, paid, suspended, err := c.register(dc, encKey, nil, acctPubB)
```

Comment:
Oh dang I was testing the loop in DiscoverAccount, and it worked well, but they going to register immediately hits the `'register'` request rate limit of once per minute.

`rpc error: error code 59: too many requests to register`

I'm considering how we want to deal with this now.  I'm thinking Register can call DiscoverAccount (have it do the loop) and use the discovered `keyIndex`...

Reply:
Gosh I really should have handled this in my last PR when I started using the `connect` route in `DiscoverAccount`.  I wanted to wrap my mind around the edges so I hacked up the idea above.

https://github.com/decred/dcrdex/commit/656507796eee694da6578a94a07b4d6192680844

It does seem to work well, and only ever calls the `register` route once.

I was able to both discover and register for a seed with two suspended accounts.

I was also able to disable the account, and then by doing the registration dialog again it discovered it without a fuss.